gas-oracle/tokenprice/tokenprice.go

```diff
@@ -135,68 +154,74 @@ func (c *Client) PriceRatioWithMode() (float64, error) {
 	return ratio, nil
 }
 
-func (c *Client) priceRatio() (float64, float64, error) {
-	ethPrice, err := c.query("ETHUSDT")
+func (c *Client) getTokenPricesFromCex() (float64, float64, error) {
+	ethPrice, err := c.queryV5(ETHUSDT)
 	if err != nil {
 		return 0, 0, err
 	}
-	mntPrice, err := c.query("BITUSDT")
+	mntPrice, err := c.queryV5(c.tokenPairForMNTPrice)
 	if err != nil {
 		return 0, 0, err
 	}
 
-	ethPrice = determineETHPrice(ethPrice)
-	mntPrice = determineMNTPrice(mntPrice)
-
-	ratio, _ := new(big.Float).Quo(ethPrice, mntPrice).Float64()
-	ratio = determineTokenRatio(ratio)
+	return mntPrice, ethPrice, nil
+}
 
-	ethPriceFloat64, _ := ethPrice.Float64()
+func (c *Client) determineMNTPrice(price float64) float64 {
+	if price > MNTPriceMax || price < MNTPriceMin {
+		return c.lastMntPrice
+	}
 
-	return ratio, ethPriceFloat64, nil
+	return price
 }
 
-func (c *Client) query(symbol string) (*big.Float, error) {
-	response, err := c.client.R().
-		SetResult(&Result{}).
-		SetQueryParams(map[string]string{
-			"symbol": symbol,
-		}).
-		Get("/spot/quote/v1/ticker/price")
-	if err != nil {
-		return nil, fmt.Errorf("cannot fetch token price result: %w", err)
-	}
-	result, ok := response.Result().(*Result)
-	if !ok {
-		return nil, fmt.Errorf("cannot parse result")
-	}
-	if result.Result.Price == "" {
-		return nil, fmt.Errorf("empty price")
+func (c *Client) determineETHPrice(price float64) float64 {
+	if price > ETHPriceMax || price < (ETHPriceMin) {
+		return c.lastEthPrice
 	}
-	bigPrice, _ := big.NewFloat(0).SetString(result.Result.Price)
-	return bigPrice, nil
+
+	return price
 }
 
-func determineMNTPrice(price *big.Float) *big.Float {
-	if price.Cmp(MNTPriceMax) == 1 || price.Cmp(MNTPriceMin) == -1 {
-		return DefaultMNTPrice
+func (c *Client) determineTokenRatio(mntPrice, ethPrice float64) float64 {
+	// calculate [tokenRatioMin, tokenRatioMax]
+	tokenRatioMin := getMax(c.lastRatio*0.95, TokenRatioMin)
+	tokenRatioMax := getMin(c.lastRatio*1.05, TokenRatioMax)
+
+	ratio := ethPrice / mntPrice
+	if ratio <= tokenRatioMin {
+		return tokenRatioMin
+	}
+	if ratio >= tokenRatioMax {
+		return tokenRatioMax
```

Comment:
Where is the default value TokenRatio (t) = 4000 (from Ming’s spec) set?

Reply:
In the code `gas-oracle/oracle/l1_client.go`, if `PriceRatioWithMode` returns an error, use the default value.